src/OrchardCore.Modules/OrchardCore.Notifications/Drivers/NotifyUserTaskDisplayDriver.cs

```diff
@@ -17,4 +24,32 @@ public class NotifyUserTaskDisplayDriver : NotifyUserTaskActivityDisplayDriver<N
         : base(htmlSanitizerService, liquidTemplateManager, notificationOptions, stringLocalizer)
     {
     }
+
+    protected override string EditShapeType { get; } = $"{typeof(NotifyUserTask).Name}_Fields_Edit";
+
+    public override async Task<IDisplayResult> UpdateAsync(NotifyUserTask model, IUpdateModel updater)
+    {
+        var viewModel = new NotifyUserTaskViewModel();
+        await updater.TryUpdateModelAsync(viewModel, Prefix);
+
+        var usernames = viewModel.Usernames?.Split(',', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries);
+
+        if (usernames?.Length == 0)
+        {
+            updater.ModelState.AddModelError(Prefix, nameof(viewModel.Usernames), S["Please provide at least one username to notify."]);
+        }
+        else
+        {
+            model.Usernames = new WorkflowExpression<string>(string.Join(',', usernames));
+        }
```

Comment:
How about having some validation here that prevents you from using more than x user names? There's a limit for `IN` expressions, [at least with SQL Server](https://stackoverflow.com/questions/21178390/in-clause-limitation-in-sql-server).

Reply:
sure. I added this.

Reply:
I don't see this being addressed.

Reply:
I still don't see this being addressed.

Reply:
I added a distinct check to reduce the number of values. We can't really write a statement to say you can only have 2 X amount of parameters because the value can be different from one server to another. It's the same reason why in the content manager we allow you to provide as many contentItemIds as you need. We don't validate at that level. Adding Distinct will help.

Reply:
OK, let's leave it like that.

Reply:
@Piedone I just made a change to query the users in chunks of 1000 per request. Happy? :)

Reply:
Did you push it too?

Reply:
https://github.com/OrchardCMS/OrchardCore/pull/16073/commits/acd9569abd3d0676289dd0edfd8dd52e7816a5d7

Reply:
Great!